src/main/kotlin/xyz/crossward/service/StatsService.kt

```diff
@@ -0,0 +1,35 @@
+package xyz.crossward.service
+
+import org.springframework.stereotype.Service
+import xyz.crossward.repository.EntryRepository
+
+@Service
+class StatsService(
+    private val entryRepository: EntryRepository
+) {
+    fun getAverageTimeByUserId(userId: String): Double {
+        val times = entryRepository.findTimesByUserId(userId)
+        var count = 0
+        var sum = 0.0
+        times.forEach {
+            it?.let {
+                count += 1
+                sum += it
+            }
+        }
+        return sum / count
+    }
+
+    fun getBestTimeByUserId(userId: String): Int {
```

Comment:
We could simplify this to use the builtin in function minOrNull() and throw an exception if there's no minimum for whatever reason
```Kotlin
fun getBestTimeByUserId(userId: String): Int =
    entryRepository.findTimesByUserId(userId).minOrNull() 
        ?: throw BadRequestException("User has no best time")
 ```